pkg/vzvmnet/vmnetinterface/vmnetinterface_darwin.go

```diff
@@ -0,0 +1,265 @@
+#cgo darwin LDFLAGS: -lobjc -framework Foundation -framework vmnet
+#include "vmnetinterface_darwin.h"
+*/
+import (
+	"C" //nolint: gocritic // dupImport: package is imported 2 times under different aliases on lines.. (gocritic)
+)
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"runtime"
+	"runtime/cgo"
+	"strings"
+	"syscall"
+	"unsafe" //nolint: gocritic // dupImport: package is imported 2 times under different aliases on lines.. (gocritic)
+
+	"github.com/Code-Hex/vz/v3/pkg/vmnet"
+	"github.com/Code-Hex/vz/v3/pkg/xpc"
+	"github.com/sirupsen/logrus"
+
+	"github.com/lima-vm/lima/v2/pkg/networks"
+	"github.com/lima-vm/lima/v2/pkg/vzvmnet"
+)
+
+// FileDescriptorForNetwork returns a file for the given vz network.
+func FileDescriptorForNetwork(ctx context.Context, vzNetwork string) (*os.File, error) {
+	// Load network configuration
+	nwCfg, err := networks.LoadConfig()
+	if err != nil {
+		return nil, err
+	}
+	vmnetConfig, ok := nwCfg.Vz[vzNetwork]
+	if !ok {
+		return nil, fmt.Errorf("networks.yaml: 'vz: %s' is not defined", vzNetwork)
+	}
+	// Request VmnetNetwork
+	network, err := vzvmnet.RequestVmnetNetwork(ctx, vzNetwork, vmnetConfig)
+	if err != nil {
+		return nil, fmt.Errorf("RequestVmnetNetwork failed: %w", err)
+	}
+
+	// Create socketpair connection as conn and file
+	conn, file, err := connAndFile()
+	if err != nil {
+		return nil, fmt.Errorf("connAndFile failed: %w", err)
+	}
+
+	iface, err := startWithNetwork(network.Raw(), xpc.NewDictionary())
+	if err != nil {
+		return nil, fmt.Errorf("startWithNetwork failed: %w", err)
+	}
+	logrus.Debugf("[vmnet] started VmnetInterface with maxPacketSize=%d, maxReadPacketCount=%d, maxWritePacketCount=%d, param=%+v",
+		iface.maxPacketSize, iface.maxReadPacketCount, iface.maxWritePacketCount, iface.param)
+
+	go func() {
+		logrus.Debug("[vmnet] vmnet interface goroutine started")
+		defer C.VmnetStopInterface(iface.ptr)
+
+		readBuffers, readVMPktDescs := NewVMPktDescArrayWithBuffers(iface.maxReadPacketCount, iface.maxPacketSize)
+		defer C.deallocateVMPktDescArray(readVMPktDescs)
+
+		defer conn.Close()
+		rawConn, _ := conn.(syscall.Conn).SyscallConn()
+
+		// Set packets available event callback to read packets from vmnet interface
+		cgoHandle := cgo.NewHandle(packetsAvailableEventCallback(func(estimatedCount C.int) {
+			for estimatedCount > 0 {
+				count := min(estimatedCount, iface.maxReadPacketCount)
+				// Reset vmPktDescs before reading
+				C.resetVMPktDescArray(readVMPktDescs, iface.maxReadPacketCount, iface.maxPacketSize)
+				// Read packets from vmnet interface
+				if result := vmnet.Return(C.VmnetRead(iface.ptr, readVMPktDescs, &count)); result != vmnet.ErrSuccess {
+					logrus.WithError(result).Error("VmnetRead failed")
+					return
+				}
+				// Write packets to the connection
+				writeBuffers := buffersForWritingToConn(readVMPktDescs, count)
+				if _, err := writeBuffers.WriteTo(conn); err != nil {
```

Comment:
Are these Lint errors false positive? https://github.com/golangci/golangci-lint/issues/3628
https://github.com/lima-vm/lima/actions/runs/20463316083/job/58800895757?pr=4394#step:6:36
> uncheckedInlineErr: hea error is unchecked, maybe intended to check it instead of ing (gocritic)

Reply:
Added ` //nolint: gocritic // https://github.com/golangci/golangci-lint/issues/3628` to each lines.